elements/lisk-codec/src/string.ts

```diff
@@ -11,20 +11,15 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
-import { writeVarInt } from './varint';
+import { readBytes, writeBytes } from './bytes';
 
-interface SchemaProperty {
-	readonly dataType: string;
-}
-
-export const writeString = (value: string, _schema: SchemaProperty): Buffer => {
+export const writeString = (value: string): Buffer => {
 	const stringBuffer = Buffer.from(value, 'utf8');
 
-	return Buffer.concat([
-		writeVarInt(stringBuffer.length, { dataType: 'uint32' }),
-		stringBuffer,
-	]);
+	return writeBytes(stringBuffer);
 };
 
-export const readString = (buffer: Buffer, _schema: SchemaProperty): string =>
-	buffer.toString('utf8', 1, buffer.length);
+export const readString = (buffer: Buffer, offset: number): [string, number] => {
+	const [value, size] = readBytes(buffer, offset);
```

Comment:
why you need readBytes here if you can just directly read it with Buffer native functions?

Reply:
same as above, it needs to read the first X bytes to get the length, and then value is obtained, which is same as bytes. Only difference with bytes is how the result Buffer is read

Reply:
I think having this separated would be easier for other readers but guess it's ok like this too.